### ISSUE TITLE
Different header results using processHeader or processFulltext

### ISSUE THREAD

**Issue by Aazhar** [kermitt2/grobid#281]
Processing document DOI 10.1007/s10676-016-9400-6 gives different output when extracting only header metadata (poor metadata, no abstract..) comparing to extracting the fulltext.

Process Fulltext Document:

    <teiHeader xml:lang="en">
        <encodingDesc>
            <appInfo>
                <application version="0.5.1-SNAPSHOT" ident="GROBID" when="2018-01-29T13:38+0000">
                    <ref target="https://github.com/kermitt2/grobid">GROBID - A machine learning software for extracting information from scholarly documents</ref>
                </application>
            </appInfo>
        </encodingDesc>
        <fileDesc>
            <titleStmt>
                <title level="a" type="main">AI assisted ethics</title>
            </titleStmt>
            <publicationStmt>
                <publisher>Springer Nature</publisher>
                <availability status="unknown">
                    <p>Copyright Springer Nature</p>
                </availability>
                <date type="published" when="2016">2016</date>
            </publicationStmt>
            <sourceDesc>
                <biblStruct>
                    <analytic>
                        <author>
                            <persName
                                xmlns="http://www.tei-c.org/ns/1.0">
                                <forename type="first">Amitai</forename>
                                <surname>Etzioni</surname>
                            </persName>
                        </author>
                        <author>
                            <persName
                                xmlns="http://www.tei-c.org/ns/1.0">
                                <forename type="first">Oren</forename>
                                <surname>Etzioni</surname>
                            </persName>
                        </author>
                        <title level="a" type="main">AI assisted ethics</title>
                    </analytic>
                    <monogr>
                        <title level="j" type="main">Ethics and Information Technology</title>
                        <title level="j" type="abbrev">Ethics Inf Technol</title>
                        <idno type="ISSN">1388-1957</idno>
                        <idno type="eISSN">1572-8439</idno>
                        <imprint>
                            <publisher>Springer Nature</publisher>
                            <biblScope unit="volume">18</biblScope>
                            <biblScope unit="issue">2</biblScope>
                            <biblScope unit="page" from="149" to="156"/>
                            <date type="published" when="2016">2016</date>
                        </imprint>
                    </monogr>
                    <idno type="DOI">10.1007/s10676-016-9400-6</idno>
                    <note>Electronic copy available at: AI assisted ethics Electronic copy available at: 1 2 3 Your article is published under the Creative Commons Attribution license which allows users to read, copy, distribute and make derivative works, as long as the author of the original work is cited. You may self-archive this article on your own website, an institutional repository or funder&apos;s repository and make it publicly available immediately. ORIGINAL PAPER Ó Springer Science+Business Media Dordrecht 2016</note>
                </biblStruct>
            </sourceDesc>
        </fileDesc>
        <profileDesc>
            <textClass>
                <keywords>
                    <term>Ethics bot</term>
                    <term>Communiterianism</term>
                    <term>Second-layer AI</term>
                    <term>Driverless cars</term>
                </keywords>
            </textClass>
            <abstract>
                <p>The growing number of &apos;smart&apos; instruments, those equipped with AI, has raised concerns because these instruments make autonomous decisions; that is, they act beyond the guidelines provided them by programmers. Hence, the question the makers and users of smart instrument (e.g., driver-less cars) face is how to ensure that these instruments will not engage in unethical conduct (not to be conflated with illegal conduct). The article suggests that to proceed we need a new kind of AI program-oversight programs-that will monitor, audit, and hold operational AI programs accountable.</p>
            </abstract>
        </profileDesc>
    </teiHeader>

Process Header Document :

    <teiHeader xml:lang="en">
        <encodingDesc>
            <appInfo>
                <application version="0.5.1-SNAPSHOT" ident="GROBID" when="2018-01-29T13:54+0000">
                    <ref target="https://github.com/kermitt2/grobid">GROBID - A machine learning software for extracting information from scholarly documents</ref>
                </application>
            </appInfo>
        </encodingDesc>
        <fileDesc>
            <titleStmt>
                <title level="a" type="main">AI assisted ethics</title>
            </titleStmt>
            <publicationStmt>
                <publisher>Springer Nature</publisher>
                <availability status="unknown">
                    <p>Copyright Springer Nature</p>
                </availability>
                <date type="published" when="2016-06">2016-06</date>
            </publicationStmt>
            <sourceDesc>
                <biblStruct>
                    <analytic>
                        <author>
                            <persName
                                xmlns="http://www.tei-c.org/ns/1.0">
                                <forename type="first">Amitai</forename>
                                <surname>Etzioni</surname>
                            </persName>
                        </author>
                        <author>
                            <persName
                                xmlns="http://www.tei-c.org/ns/1.0">
                                <forename type="first">Oren</forename>
                                <surname>Etzioni</surname>
                            </persName>
                        </author>
                        <author>
                            <affiliation>
                                <orgName>1 2 3 Ethics and Information Technology</orgName>
                            </affiliation>
                        </author>
                        <title level="a" type="main">AI assisted ethics</title>
                    </analytic>
                    <monogr>
                        <title level="j" type="main">Ethics and Information Technology</title>
                        <title level="j" type="abbrev">Ethics Inf Technol</title>
                        <idno type="ISSN">1388-1957</idno>
                        <idno type="eISSN">1572-8439</idno>
                        <imprint>
                            <publisher>Springer Nature</publisher>
                            <biblScope unit="volume">18</biblScope>
                            <biblScope unit="issue">2</biblScope>
                            <biblScope unit="page" from="149" to="156"/>
                            <date type="published" when="2016-06" />
                        </imprint>
                    </monogr>
                    <idno type="DOI">10.1007/s10676-016-9400-6</idno>
                    <note>Electronic copy available at: AI assisted ethics Amitai Etzioni &amp; Oren Etzioni</note>
                </biblStruct>
            </sourceDesc>
        </fileDesc>
        <profileDesc>
            <abstract/>
        </profileDesc>
    </teiHeader>

**Comment by kermitt2** [comment]
This is not a bug, this is the expected current behaviour :) 

When processing the whole full text, the header segmentation is more accurate because the zone segmentation is working on the whole article. When processing only header, only the first two pages are considered which speed-up considerably the extraction (because it's pdf2xml which takes a lot of time) - a factor 10, for a loss of accuracy in practice quite negligible according to my benchmarks.  

To summarize, in general the processFulltext results for header metadata might be slightly more accurate (by 0.5-1.5% f-score), but processing the whole full text takes a factor 10 in time processing as compared to processHeader.

**Comment by kermitt2** [comment]
The heuristics has disappeared from 0.6.1 version (at least by default, there's still some unused stuff for the header processing that I will remove in the future). 
There is now only one process for both `processHeader` and `processFulltext` using the segmentation model for the header zone segmentation, either on 2 first pages for `processHeader` or on the full documents for  `processFulltext`. Normally the only case where there could be some differences now is when the header goes beyond the 2 first pages.
